components/camera/videosource/webcam.go

```diff
@@ -290,10 +293,14 @@ func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logge
 	if err != nil {
 		return nil, errors.Wrap(err, "cannot find video source for camera")
 	}
+	label := getLabelFromCamera(utils.UnwrapProxy(cam).(camera.Camera), logger)
 
-	label := attrs.Path
-	if label == "" {
-		label = getLabelFromCamera(utils.UnwrapProxy(cam).(camera.Camera), logger)
+	// grab this camera's label from the registry if it's there
+	foundCamLabel, ok := camRegistry[attrs.Path]
+	if !ok {
+		camRegistry[attrs.Path] = label
+	} else {
+		label = foundCamLabel
```

Comment:
Test this out on macOS, and see if the paths still work as expected for that platform.

Reply:
Kinda blocked on macOS testing by (https://viam.atlassian.net/browse/RSDK-1790). Maybe we just consider this a Linux fix for now?